aws/external/config_test.go

```diff
@@ -101,4 +103,13 @@ func TestConfigs_ResolveAWSConfig(t *testing.T) {
 	if e, a := "provider", creds.Source; e != a {
 		t.Errorf("expect %v provider, got %v", e, a)
 	}
+
+	var expectedSources []interface{}
+	for _, s := range cfg.ConfigSources {
+		expectedSources = append(expectedSources, s)
+	}
+
+	if e, a := expectedSources, cfg.ConfigSources; !reflect.DeepEqual(e, a) {
+		t.Errorf("expected %v, got %v", e, a)
```

Comment:
QQ: When to use `reflect.DeepEqual` vs `cmp.Diff`?

I thought cmp.Diff was preferred, but for []interface not sure

Reply:
I think this is a much bigger discussion we can take offline, but in essence cmp.Diff has much more granular comparisons, for example know that two timestamps are equal across timezones. Additionally it has the ability to set options like ignoring/not-ignoring unexported structure fields. In this instance of comparison the need for using cmp.Diff is unwarranted.